sqlgen/expression_sql_generator.go

```diff
@@ -273,7 +273,12 @@ func (esg *expressionSQLGenerator) literalTime(b sb.SQLBuilder, t time.Time) {
 		esg.placeHolderSQL(b, t)
 		return
 	}
-	esg.Generate(b, t.Format(esg.dialectOptions.TimeFormat))
+	tz, err := time.LoadLocation(timeLocation)
```

Comment:
Perhaps loading the location should happen on the `SetTimeLocation` call? That way it could return the error directly. WDYT?

Reply:
Instead of loading it in `goqu` I changed `SetTimeLocation` to accept a `*time.Location`. I think it makes it clearer what `goqu` is using to convert the time.